operator/src/main/java/oracle/kubernetes/weblogic/domain/model/ClusterStatus.java

```diff
@@ -174,6 +174,20 @@ public Long getObservedGeneration() {
     return observedGeneration;
   }
 
+  /**
+   * The observedGeneration attribute is defined as an integer type in the Cluster Resource
+   * schema, hence convert to integer when publishing the status.
+   * @return integer value of the observedGeneration, if set, otherwise defaults to 1.
+   */
+  private Integer getObservedGenerationAsInteger() {
+    return observedGeneration != null ? observedGeneration.intValue() : 1;
```

Comment:
why is the default value for observedGeneration 1? Thanks

Reply:
Since the Observed Generation is a monotonically increasing number that gets bumped every time you update your resource. So the first resource is created with generation 1 by 'metadata.Generation', hence, I default to 1.  In most cases, we'll have read the ClusterResource object from the API server when listing them and the Generation will have been set by Kubernetes.

bash-4.2$ kubectl describe clu sample-domain1-cluster-1 -n sample-domain1-ns
Name:         sample-domain1-cluster-1
Namespace:    sample-domain1-ns
Labels:       weblogic.domainUID=sample-domain1
Annotations:  <none>
API Version:  weblogic.oracle/v1
Kind:         Cluster
Metadata:
  Creation Timestamp:  2022-09-28T18:33:47Z
  Generation:          1
  Managed Fields:
    API Version:  weblogic.oracle/v1
    Fields Type:  FieldsV1
...
    Manager:         Kubernetes Java Client
    Operation:       Update
    Subresource:     status
    Time:            2022-09-28T18:36:08Z
  Resource Version:  215489
  UID:               618d8ee1-f0cc-4f5a-85a3-4e273fd152dc
Spec:
  Cluster Name:  cluster-1
  Replicas:      2
Status:
  Cluster Name:  cluster-1
  Conditions:
    Last Transition Time:  2022-09-28T18:37:52.312865Z
    Status:                True
    Type:                  Available
    Last Transition Time:  2022-09-28T18:37:52.312918Z
    Status:                True
    Type:                  Completed
  Label Selector:          weblogic.domainUID=sample-domain1,weblogic.clusterName=cluster-1
  Maximum Replicas:        5
  Minimum Replicas:        0
  Observed Generation:     1
  Ready Replicas:          2
  Replicas:                2
  Replicas Goal:           2